perfkitbenchmarker/providers/azure/azure_virtual_machine.py

```diff
@@ -198,9 +198,10 @@ def _Create(self):
     create_cmd = [AZURE_PATH,
                   'vm',
                   'create',
-                  '--affinity-group=%s' % self.network.affinity_group.name,
+                  '--location=%s' % self.zone,
                   '--virtual-network-name=%s' % self.network.vnet.name,
                   '--vm-size=%s' % self.machine_type,
+                  '--public-ip=%s' % self.name,
```

Comment:
Each VM in Azure comes automatically with a public and private IP address. The public (VIP) is not really associated with the VM but to the cloud service account that the VM lives in. By default, VIP cannot be pinged at all. 

If we do not utilize the public IP address, we would not be able to test latency across regions either on the VIP or private IP. Caveat is if we setup vnet-to-vnet VPN but it's out of scope of this test module. This contrasts with GCP where a subnet can be stretched across all of GPC's compute region. 

I'm proposing that we open up the ping test to allow cross-region ping test in addition to inter-datacenter ping.